.github/workflows/docs.yaml

```diff
@@ -15,10 +15,51 @@ jobs:
       - uses: actions/setup-python@v2
         with:
           python-version: '3.9'
+      - uses: ruby/setup-ruby@v1.160.0
+        with:
+          ruby-version: '3.2'
+
       - name: Install MkDocs
         run: pip install -r docs/doc_requirements.txt
-      - name: Publish docs
+      - name: Build legacy mkdocs documentation
+        run: make mkdocs
+
+      - name: Install Asciidoc
+        run: make docs-dependencies
+      - name: Build user docs
+        run: make docs-user
+      - name: Build dev docs
+        run: make docs-dev
+
+      - name: Prepare gh-pages branch
         run: |
           git config user.name github-actions
           git config user.email github-actions@github.com
-          mkdocs gh-deploy --force --clean --verbose
+
+          git checkout -b gh-pages aabb4803f89a3b006094d8ae7a933650f644d036
```

Comment:
I should probably put here a gh-pages branch delete here to be sure. I guess only 'main' branch gets created locally and the 'gh-pages' is only remote so that's why it wasn't complaining in my testing and would probably work, but i'll make it explicit.